admin/settings/class-admin-apple-settings-section.php

```diff
@@ -677,4 +650,62 @@ public function get_section_info() {
 		return '';
 	}
 
+	/**
+	 * Sanitizes a single dimension array with text values.
+	 *
+	 * @param array $value
+	 * return array
+	 */
+	public function sanitize_array( $value ) {
+		return array_map( 'sanitize_text_field', $value );
+	}
+
+	/**
+	 * Get the current value for an option.
+	 */
+	public function get_value( $key ) {
+		$saved_settings = get_option( $this->option_name );
+		return ( ! empty( $saved_settings[ $key ] ) ) ? $saved_settings[ $key ] : '';
+	}
+
+	/**
+	 * Each section is responsible for saving its own settings
+	 * since only it knows the nature of the fields and sanitization methods.
+	 */
+	public function save_settings() {
+		// Check if we're saving options and that there are settings to svae
+		if ( empty( $_POST['action'] )
+			|| 'apple_news_options' !== $_POST['action']
+			|| empty( $this->settings ) ) {
```

Comment:
I pointed this out because `$this->settings` is filtered immediately before this method is called (`$this->settings = apply_filters( 'apple_news_section_settings', $this->settings, $page );`), so even though it may be unlikely, it's not impossible.